### DISCUSSION TITLE
Talking to the child outlet from root.ts to get info

### DISCUSSION THREAD

**Discussion by kof** [remix-run/remix#2030] [Q&A] [answered]
I can see only 2 options that would allow me to set a class on the body based on the route I am at, given that the name of the class is only known in the subroute module, because its dynamic, e.g. being fetched from a database.

1. Have a second root2.tsx for that route and have it fetch in loader the info
2. Talk to the nested child route in some way to get the info

So far I couldn't find a good way to do it, what am I missing?

**Comment by sergiodxa** [comment, marked as answer]
You can use the useMatches hook to get the data returned by any route, including child routes.

**Comment by kof** [comment]
Yeah I saw this possibility, but feels like a hack, misused api, what I really want is an official way to use a different root.tsx

**Reply by kof** [reply]
Created this issue now https://github.com/remix-run/remix/discussions/2039

**Reply by kiliman** [reply]
1) `useMatches` is the documented way to share route data, so it's not a hack.
2) There is only one *root* route. You can simply render an `<Outlet/>` then have differently layout routes based on current route.
3) It's still a React app, so you can use any React methods to pass data to a parent from a child. For example use React Context that returns a setter function.

**Reply by kof** [reply]
What do you mean by 2? I can't render an Outlet in a different layout, my parent layout is always going to be the root component, unless I am missing something

**Reply by kiliman** [reply]
```ts
// root.tsx
export default function Root() {
  return <Outlet/>
}
// __layout1.tsx
export default function Layout1() {
  return <Document><Outlet/></Document>
}
// __layout1/page1.tsx
export default function Page1() {
  return <h1>This will show up in layout1 outlet</h1>
}
// __layout2.tsx
export default function Layout2() {
  return <Document><Outlet/></Document>
}
// __layout2/page2.tsx
export default function Page2() {
  return <h1>This will show up in layout2 outlet</h1>
}

**Reply by kof** [reply]
Oh, interesting, thank you! That looks like the solution I was looking for and didn't realize that I could move Root contents into different child Layouts and keep Root essentially as an empty container, neat!

**Reply by kiliman** [reply]
Remember, all an `<Outlet/>` does is specify the place where its child route should be rendered. Kind of like using the `{children}` prop in React.

So by simply returning `<Outlet/>` in the *root*, you're pushing the page layout responsibility to its child layouts, where you can now have multiple.

**Reply by kiliman** [reply]
A common use for this is different layouts for *public* vs *protected* routes.

**Reply by kof** [reply]
Got it, I just didn't imagine for some reason moving Root's definitions into an Outlet, as if Root is a special place to be lol

**Reply by kiliman** [reply]
The only distinction of *root* is that it's at the top of the route tree. Nothing else.

**Reply by kof** [reply]
Yep, I was mentally blocked on this realization because its called Root and not Outlet or RootOutlet or something
